src/common/AuthContext.jsx

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable react/jsx-no-constructed-context-values */
+import { createContext, useState, useContext } from 'react';
+import PropTypes from 'prop-types';
+
+const AuthContext = createContext({
+  currentUser: null,
+  setCurrentUser: () => {},
+});
+
+const useAuthContext = () => {
+  return useContext(AuthContext);
+};
+
+const AuthContextProvider = ({ children }) => {
+  const [currentUser, setCurrentUser] = useState(null);
+  return (
+    <AuthContext.Provider value={{ currentUser, setCurrentUser }}>{children}</AuthContext.Provider>
+  );
+};
+
+AuthContextProvider.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+export { useAuthContext, AuthContextProvider };
```

Comment:
⚠️ **[eslint]** <react-refresh/only-export-components> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components.